src/CpAmm.ts

```diff
@@ -973,175 +977,115 @@ export class CpAmm {
       currentSlot,
       inputTokenInfo,
       outputTokenInfo,
+      tokenADecimal,
+      tokenBDecimal,
+      hasReferral,
     } = params;
-    const {
-      sqrtPrice: sqrtPriceQ64,
-      liquidity: liquidityQ64,
-      activationType,
-      activationPoint,
-      collectFeeMode,
-      poolFees,
-    } = poolState;
-    const {
-      feeSchedulerMode,
-      cliffFeeNumerator,
-      numberOfPeriod,
-      reductionFactor,
-      periodFrequency,
-    } = poolFees.baseFee;
-    const dynamicFee = poolFees.dynamicFee;
-
-    let actualAmountIn = inAmount;
-    if (inputTokenInfo) {
-      actualAmountIn = calculateTransferFeeExcludedAmount(
-        inAmount,
-        inputTokenInfo.mint,
-        inputTokenInfo.currentEpoch
-      ).amount;
-    }
+    const { activationType } = poolState;
+
     const aToB = poolState.tokenAMint.equals(inputTokenMint);
-    const currentPoint = activationType ? currentTime : currentSlot;
 
-    let dynamicFeeParams: DynamicFeeParams;
-    if (dynamicFee.initialized) {
-      const { volatilityAccumulator, binStep, variableFeeControl } = dynamicFee;
-      dynamicFeeParams = { volatilityAccumulator, binStep, variableFeeControl };
-    }
+    const currentPoint = activationType
```

Comment:
i think, to more clearly it should be check activationType is slot/timestamp (ack that activationType is 1 is timestamp but need more clearly)